provider/cluster/cluster_resource.go

```diff
@@ -0,0 +1,706 @@
+}
+
+func (r *ClusterResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "OpenShift managed cluster.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description: "Unique identifier of the cluster.",
+				Computed:    true,
+			},
+			"product": schema.StringAttribute{
+				Description: "Product ID OSD or Rosa",
+				Required:    true,
+			},
+			"name": schema.StringAttribute{
+				Description: "Name of the cluster.",
+				Required:    true,
+			},
+			"cloud_provider": schema.StringAttribute{
+				Description: "Cloud provider identifier, for example 'aws'.",
+				Required:    true,
+			},
+			"cloud_region": schema.StringAttribute{
+				Description: "Cloud region identifier, for example 'us-east-1'.",
+				Required:    true,
+			},
+			"multi_az": schema.BoolAttribute{
+				Description: "Indicates if the cluster should be deployed to " +
+					"multiple availability zones. Default value is 'false'.",
+				Optional: true,
+				Computed: true,
+				PlanModifiers: []planmodifier.Bool{
+					boolplanmodifier.RequiresReplace(),
+				},
+			},
+			"properties": schema.MapAttribute{
+				Description: "User defined properties.",
+				ElementType: types.StringType,
+				Optional:    true,
+				Computed:    true,
+			},
+			"api_url": schema.StringAttribute{
+				Description: "URL of the API server.",
+				Computed:    true,
+			},
+			"console_url": schema.StringAttribute{
+				Description: "URL of the console.",
+				Computed:    true,
+			},
+			"compute_nodes": schema.Int64Attribute{
+				Description: "Number of compute nodes of the cluster.",
+				Optional:    true,
+				Computed:    true,
+			},
+			"compute_machine_type": schema.StringAttribute{
+				Description: "Identifier of the machine type used by the compute nodes, " +
+					"for example `r5.xlarge`. Use the `ocm_machine_types` data " +
+					"source to find the possible values.",
+				Optional: true,
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"ccs_enabled": schema.BoolAttribute{
+				Description: "Enables customer cloud subscription.",
+				Optional:    true,
+				Computed:    true,
+			},
+			"aws_account_id": schema.StringAttribute{
+				Description: "Identifier of the AWS account.",
+				Optional:    true,
+			},
+			"aws_access_key_id": schema.StringAttribute{
+				Description: "Identifier of the AWS access key.",
+				Optional:    true,
+				Sensitive:   true,
+			},
+			"aws_secret_access_key": schema.StringAttribute{
+				Description: "AWS access key.",
+				Optional:    true,
+				Sensitive:   true,
+			},
+			"aws_subnet_ids": schema.ListAttribute{
+				Description: "AWS subnet IDs.",
+				ElementType: types.StringType,
+				Optional:    true,
+				PlanModifiers: []planmodifier.List{
+					listplanmodifier.RequiresReplace(),
```

Comment:
It wasn't set as RequiresReplace - Just to make sureyou meant for that

Reply:
Took from rosa classic and i assume it is right